lib/fluent/plugin/buffer.rb

```diff
@@ -444,6 +451,12 @@ def write(metadata_and_data, format: nil, size: nil, enqueue: false)
             end
             chunk.mon_exit rescue nil # this may raise ThreadError for chunks already committed
           end
+          if @delay_buffer_chunk_overflow_error
+            # Notify delayed chunk overflow here
+            message = @delay_buffer_chunk_overflow_error
+            @delay_buffer_chunk_overflow_error = nil
```

Comment:
This buffer should be handled by multiple output threads.
I believe this implementation covered for most cases but we should block other threads with `synchronize` blocks.

Reply:
Thanks, fixed.